test/kitchen_sink/CMakeLists.txt

```diff
@@ -85,7 +86,7 @@ target_compile_options(kitchen_sink_options INTERFACE
         )
 
 target_compile_definitions(kitchen_sink_libs INTERFACE
-        NDEBUG
+        PARAM_ASSERTIONS_ENABLE_ALL=1 # want to check all the assertions for compilation warnings
```

Comment:
Haha, I'd never noticed that this file is manually specifying `NDEBUG`! That explains why I couldn't get the compiler to throw errors at me when I deliberately put syntax-errors inside `valid_params_if` / `invalid_params_if` macros :rofl:

Reply:
yup; i have no idea why that was like that :-)